src/core/fragment.jl

```diff
@@ -667,6 +673,205 @@ end
+- `Bool`: `true` if the fragment has a torsion angle phi, `false` otherwise.
+"""
+@inline function has_torsion_phi(frag::Fragment)
+    chain = parent_chain(frag)
+    if nresidues(chain) < 2
+        return false
+    end
+    return !has_flag(frag, :N_TERMINAL) && is_amino_acid(frag)
+end
+
+"""
+    has_torsion_omega(frag::Fragment) -> Bool
+
+Checks if the given `Fragment` object has a torsion angle omega defined.
+    See [Wikipedia](https://en.wikipedia.org/wiki/File:Protein_backbone_PhiPsiOmega_drawing.svg) for more information.
+# Arguments
+- `frag::Fragment`: The fragment object to be checked.
+
+# Returns
+- `Bool`: `true` if the fragment has a torsion angle omega, `false` otherwise.
+"""
+@inline function has_torsion_omega(frag::Fragment)
+    chain = parent_chain(frag)
+    if nresidues(chain) < 2
+        return false
+    end
+    return !has_flag(frag, :N_TERMINAL) && is_amino_acid(frag)
+end
+
+
+"""
+    calculate_torsion_angle(a::Atom{T}, b::Atom{T}, c::Atom{T}, d::Atom{T}) -> T
+
+Calculates the torsion angle (dihedral angle) defined by four atoms `a`, `b`, `c`, and `d`.
+The torsion angle is the angle between the plane formed by atoms `a`, `b`, `c` and the
+plane formed by atoms `b`, `c`, `d`.
+
+# Arguments
+- `a::Atom`: The first atom in the sequence.
+- `b::Atom`: The second atom in the sequence.
+- `c::Atom`: The third atom in the sequence.
+- `d::Atom`: The fourth atom in the sequence.
+
+# Returns
+- `Float64`: The torsion angle in radians.
+"""
+function calculate_torsion_angle(a::Atom{T}, b::Atom{T}, c::Atom{T}, d::Atom{T}) where T
+
+    n12 = cross(b.r-a.r, c.r-b.r)
+    n34 = cross(c.r-b.r, d.r-c.r)
+
+    if iszero(n12) || iszero(n34)
+        @warn "Illegal positions"
+    end
+
+    n12 /= norm(n12)
+    n34 /= norm(n34)
+
+    scalar_product = clamp(dot(n12,n34), -one(T), one(T))
+    #take the direction into account direction = dot(cross(n12,n34), (c.r-b.r))
+    return dot(cross(n12,n34), (c.r-b.r)) < zero(T) ? -one(T) * acos(scalar_product) : acos(scalar_product)
+end
+
+"""
+    set_torsion_angle!(a::Atom, b::Atom, c::Atom, d::Atom, angle::Union{Float32, Float64})
+
+Sets the torsion angle defined by the four atoms `a`, `b`, `c`, and `d` to the specified `angle`.
+
+# Arguments
+- `a::Atom{T}`: The first atom in the torsion angle definition.
+- `b::Atom{T}`: The second atom in the torsion angle definition.
+- `c::Atom{T}`: The third atom in the torsion angle definition.
+- `d::Atom{T}`: The fourth atom in the torsion angle definition.
+- `angle::{T}`: The desired torsion angle in radians.
+
+# Returns
+- `Bool`: `true` if the torsion angle has been set successfully, `false` otherwise.
+
+!!! note
+This function modifies the positions of the atoms to achieve the specified torsion angle. It
+assumes that the atoms are part of a molecular structure where such modifications are meaningful. It does not
+check for steric clashes or other geometric or chemical constraints that may arise from the rotation.
```

Comment:
```suggestion
    This function modifies the positions of the atoms to achieve the specified torsion angle. It
    assumes that the atoms are part of a molecular structure where such modifications are meaningful. It does not
    check for steric clashes or other geometric or chemical constraints that may arise from the rotation.
```
The indentation is required here for the text to be included in the "note" box.